From: Herbert Xu <herbert@gondor.apana.org.au>
Date: Wed, 2 Nov 2005 07:55:00 +0000 (+1100)
Subject:     [NET]: Fix zero-size datagram reception
X-Git-Url: http://www.kernel.org/git/gitweb.cgi?p=linux/kernel/git/torvalds/linux-2.6.git;a=commitdiff;h=c75d721c761ad0f2d8725c40af9e4f376efefd24

  [NET]: Fix zero-size datagram reception
  
  The recent rewrite of skb_copy_datagram_iovec broke the reception of
  zero-size datagrams.  This patch fixes it.
  
  Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
  Signed-off-by: Arnaldo Carvalho de Melo <acme@mandriva.com>
---

--- a/net/core/datagram.c
+++ b/net/core/datagram.c
@@ -213,6 +213,10 @@ int skb_copy_datagram_iovec(const struct
 {
 	int i, err, fraglen, end = 0;
 	struct sk_buff *next = skb_shinfo(skb)->frag_list;
+
+	if (!len)
+		return 0;
+
 next_skb:
 	fraglen = skb_headlen(skb);
 	i = -1;
